module-1/task-2/index.js

```diff
@@ -0,0 +1,45 @@
+const { pipeline, Transform } = require('stream');
+const fs = require('fs');
+const csv = require('csvtojson');
+
+const PATH_TO_CSV = './module-1/task-2/csv/sample.csv';
+const PATH_TO_TXT = './module-1/task-2/txt/sample.txt';
+
+const transformFn = (rawObj) => {
+  const keysToOmit = ['amount'];
+  const keysToConvertToNumber = ['price'];
+  const rawObjCopy = { ...rawObj };
+
+  return Object.keys(rawObjCopy)
+    .filter((key) => !keysToOmit.includes(key.toLowerCase()))
+    .reduce((newObj, key) => {
+      newObj[key.toLowerCase()] = keysToConvertToNumber.includes(key.toLowerCase())
+        ? parseFloat(rawObjCopy[key])
+        : rawObjCopy[key];
+
+      return newObj;
+    }, {});
+};
+
+const main = () => {
+  pipeline(
+    fs.createReadStream(PATH_TO_CSV),
+    csv(),
+    new Transform({
```

Comment:
By using Transform here you're essentially storing parts of the stream as Buffers (aka in RAM memory), violating the task requirements.
Moreover, it's a bit unclear why would you need to use transform to accomplish the task that is already handled by csv() method.
The case sensitiveness issue is cause by using quotes around every value in csv file, which is note the case in a typical csv file (the content is already a string)

Reply:
So I guess I overcomplicated this thing a little bit. Removed the transformation at all, seems to work as expected. However what if I really need to transform the resulting object somehow? How should that be done not to bloat RAM memory etc? As far as I remember at first I used [.subscribe](https://github.com/Keyang/node-csvtojson#result-transform) method from csv library itself but it didn't work, the code inside this method just never got executed, I assume because calling `csv()` just returned a stream downwards. Then I switched to using `Transform` and that seemed to solve the issue

Reply:
So I guess I overcomplicated this thing a little bit. Removed the transformation at all, seems to work as expected. However what if I really need to transform the resulting object somehow? How should that be done not to bloat RAM memory etc? As far as I remember at first I used [.subscribe](https://github.com/Keyang/node-csvtojson#result-transform) method from csv library itself but it didn't work, the code inside this method just never got executed, I assume because calling `csv()` just returned a stream downwards. Then I switched to using `Transform` and that seemed to solve the issue

Reply:
In general, you're correct in your assumption that if a transform is needed you need to store at least something in the memory to perform the necessary operations. 
While streaming data you're basically breaking it into "chunks". Depending on the type of the transformation and the data itself you either perform the operations on the whole data (once its chunks are all in memory) or on the individual chunks.
`data = a + b` => `T*data = T1*a + T2*b` OR `T*data = T0*(a + b)`;
To put it simply: 
If the data is homogeneous (an image) - you can transform them chunk by chunk and the memory consumption stays low (like image resizing). 
If the data is heterogeneous - you'll need to collect the whole thing and apply processing.

Reply:
But the task was simple - just stream data from one place to another (one format to another) not really caring about its content :)